packages/modules/vehicles/smarteq/api.py

```diff
@@ -0,0 +1,262 @@
+        return code
+
+    # get Tokens
+    def get_Tokens(self, code: str) -> dict:
+        url5 = TOKEN_URL
+        headers5 = {
+            "Accept": "*/*",
+            "User-Agent": "sOAF/202108260942 CFNetwork/978.0.7 Darwin/18.7.0",
+            "Accept-Language": ACCEPT_LANGUAGE,
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        data5 = "grant_type=authorization_code&code=" + code + "&code_verifier=" + self.code_verifier +\
+                "&redirect_uri=" + REDIRECT_URI + "&client_id=" + CLIENT_ID
+
+        try:
+            response5 = self.session.post(url5, headers=headers5, data=data5)
+            # self.log.debug("response5.status_code = " + str(response5.status_code))
+            Tokens = json.loads(response5.text)
+            if not Tokens['access_token']:
+                self.log.warn("no access_token found")
+                return {}
+            # self.log.debug("Tokens=\n" + json.dumps(Tokens, indent=4))
+        except Exception as e:
+            self.log.exception("Step5 Exception: " + str(e))
+        return Tokens
+
+    # reconnect to Server
+    def reconnect(self) -> dict:
+        token = self.login()
+        if not token:
+            self.log.error("Login failed, token empty - Abort")
+            return {}
+
+        code = self.get_code(token)
+        self.Tokens = self.get_Tokens(code)
+
+        tf = open(self.tokensFile, "wb")
+        pickle.dump(self.Tokens, tf)
+        tf.close()
+
+        return self.Tokens
+
+    # get_status -> soc, range of Vehicle
+    def get_status(self, vin: str) -> Union[int, float]:
+        url7 = STATUS_URL + "/seqc/v0/vehicles/" + vin +\
+               "/init-data?requestedData=BOTH&countryCode=DE&locale=de-DE"
+        headers7 = {
+            "accept": "*/*",
+            "accept-language": "de-DE;q=1.0",
+            "authorization": "Bearer " + self.Tokens['access_token'],
+            "x-applicationname": CLIENT_ID,
+            "user-agent": "Device: iPhone 6; OS-version: iOS_12.5.1; App-Name: smart EQ control; App-Version: 3.0;\
+            Build: 202108260942; Language: de_DE",
+            "guid": GUID
+        }
+
+        try:
+            response7 = self.session.get(url7, headers=headers7)
+            # self.log.debug("response7.status_code = " + str(response7.status_code))
+            res7 = json.loads(response7.text)
+            soc = int(res7['precond']['data']['soc']['value'])
+            range = float(res7['precond']['data']['rangeelectric']['value'])
+        except Exception as e:
+            self.log.exception("Step7 Exception: " + str(e))
+            res7s = json.dumps(res7, indent=4)
+            self.log.error("\nget_status result for vin: " + vin + "\n" + res7s)
+            soc = -1
+            range = 0.0
+            if "Vehicle not found" in res7s:
+                soc = -2
+        return soc, range
+
+    async def _fetch_soc(self,
```

Comment:
Warum hast Du die Methode mit dem Keyword async versehen? Hier wird doch gar nichts asynchron ausgeführt.
Die einzelnen SoC-Modul werden übrigens in Threads abgefragt.